R/utilities-writing-report.R

```diff
@@ -216,38 +217,23 @@ renderWordReport <- function(fileName, intro = NULL, createWordReport = FALSE, w
   figureContent <- NULL
   for (lineContent in fileContent) {
     firstElement <- getFirstLineElement(lineContent)
-    # When finding a line referencing a table caption,
-    if (grepl(pattern = "Table", x = firstElement)) {
-      # The new content to write in the report is
-      # - previous content = wordFileContent
-      # - page break = "\\newpage"
-      # - table caption = lineContent
-      # - line space = "" (due to sep="\n" in function write)
+    # When finding a line referencing a table or figure caption
+    # The new content to write in the report is
+    # - previous content = wordFileContent
+    # - page break = "\\newpage"
+    # - caption = lineContent
+    # - line space = "" (due to sep="\n" in function write)
+    # - go on with artifact (figure/table)
+    isArtifactCaption <- any(
+      grepl(pattern = "Table", x = firstElement),
+      grepl(pattern = "Figure", x = firstElement)
```

Comment:
what happens if the report has just a text line starting with "Table " or "Figure "?

Reply:
>Would it be better to use instead the identification of the anchor that has a "table"/"figure" as id ?

yes. the probability of having "figure" or "table" within the text is not that low.